{"version":3,"file":"NFTErrorBoundary.js","sources":["../../../src/nft/components/NFTErrorBoundary.tsx"],"sourcesContent":["import {\n  Component,\n  type ComponentType,\n  type ErrorInfo,\n  type ReactNode,\n} from 'react';\n\ntype Props = {\n  fallback?: ComponentType<{ error: Error }>;\n  children: ReactNode;\n};\n\ntype State = {\n  error: Error | null;\n};\n\nclass NFTErrorBoundary extends Component<Props, State> {\n  state: State = {\n    error: null,\n  };\n\n  static getDerivedStateFromError(error: Error): State {\n    return { error };\n  }\n\n  componentDidCatch(error: Error, errorInfo: ErrorInfo) {\n    console.error('Uncaught error:', error, errorInfo);\n  }\n\n  render() {\n    if (this.state.error) {\n      if (this.props.fallback) {\n        const Fallback = this.props.fallback;\n        return <Fallback error={this.state.error} />;\n      }\n      return <h1>Sorry, we had an unhandled error</h1>;\n    }\n\n    return this.props.children;\n  }\n}\n\nexport default NFTErrorBoundary;\n"],"names":[],"mappings":";;AAgBA,MAAM,yBAAyB,UAAwB;AAAA,EAAvD,cAAA;AAAA,UAAA,GAAA,SAAA;AACE,SAAA,QAAe;AAAA,MACb,OAAO;AAAA,IAAA;AAAA,EACT;AAAA,EAEA,OAAO,yBAAyB,OAAqB;AACnD,WAAO,EAAE,MAAA;AAAA,EACX;AAAA,EAEA,kBAAkB,OAAc,WAAsB;AACpD,YAAQ,MAAM,mBAAmB,OAAO,SAAS;AAAA,EACnD;AAAA,EAEA,SAAS;AACP,QAAI,KAAK,MAAM,OAAO;AACpB,UAAI,KAAK,MAAM,UAAU;AACvB,cAAM,WAAW,KAAK,MAAM;AAC5B,eAAO,uBAAC,UAAA,EAAS,OAAO,KAAK,MAAM,MAAA,GAA5B,QAAA,OAAA;AAAA,UAAA,UAAA;AAAA,UAAA,YAAA;AAAA,UAAA,cAAA;AAAA,QAAA,GAAA,IAAmC;AAAA,MAC5C;AACA,aAAO,uBAAC,QAAG,UAAA,mCAAA,GAAJ,QAAA,OAAA;AAAA,QAAA,UAAA;AAAA,QAAA,YAAA;AAAA,QAAA,cAAA;AAAA,MAAA,GAAA,IAAoC;AAAA,IAC7C;AAEA,WAAO,KAAK,MAAM;AAAA,EACpB;AACF;"}